caffe2/utils/filler.h

```diff
@@ -41,8 +43,17 @@ class TensorFiller {
         auto fixed_sum = static_cast<Type>(fixed_sum_);
         CAFFE_ENFORCE_LE(min * tensor->numel(), fixed_sum);
         CAFFE_ENFORCE_GE(max * tensor->numel(), fixed_sum);
-        math::RandFixedSum<Type, Context>(
-            tensor->numel(), min, max, fixed_sum_, data, context);
+        auto average = fixed_sum / tensor->numel();
```

Comment:
Either we could have a new distribution in `math_cpu.cc` or incorporate this logic into `RandFixedSum`? Ideally we would avoid making these branches more complicated. (Check the TODO "replace filler distribution enum with a better abstraction" above).

Reply:
This is pretty hacky and not working for all the cases. I just want to see how much we can get while keeping the current memoryless filler implementation. I'm getting less and less confident...